terraform/keycloak/keycloak-dev/main.tf

```diff
@@ -0,0 +1,141 @@
+  realm_name             = local.bcsc_realm_name
+  standard_realm_name    = local.standard_realm_name
+  bcsc_keycloak_url      = var.bcsc_keycloak_url
+
+  depends_on = [
+    module.azureidir
+  ]
+}
+
+module "bceidbasic" {
+  source                     = "github.com/bcgov/social-access-portal-terraform-modules?ref=3066d73/modules/base-realms/realm-bceidbasic"
+  keycloak_url               = var.keycloak_url
+  realm_name                 = local.bceidbasic_realm_name
+  standard_realm_name        = local.standard_realm_name
+  saml_entity_id             = "https://dev.loginproxy.gov.bc.ca/auth/realms/_bceidbasic/"
+  single_sign_on_service_url = local.siteminder_single_sign_on_service_url
+  signing_certificate        = var.siteminder_signing_certificate
+
+  depends_on = [
+    module.bcsc
+  ]
+
+}
+
+
+module "bceidbusiness" {
+  source                     = "github.com/bcgov/social-access-portal-terraform-modules?ref=3066d73/modules/base-realms/realm-bceidbusiness"
+  keycloak_url               = var.keycloak_url
+  realm_name                 = local.bceidbusiness_realm_name
+  standard_realm_name        = local.standard_realm_name
+  saml_entity_id             = "https://dev.loginproxy.gov.bc.ca/auth/realms/_bceidbusiness/"
+  single_sign_on_service_url = local.siteminder_single_sign_on_service_url
+  signing_certificate        = var.siteminder_signing_certificate
+
+  depends_on = [
+    module.bceidbasic
+  ]
+
+}
+
+module "bceidboth" {
+  source                     = "github.com/bcgov/social-access-portal-terraform-modules?ref=3066d73/modules/base-realms/realm-bceidboth"
+  keycloak_url               = var.keycloak_url
+  realm_name                 = local.bceidboth_realm_name
+  standard_realm_name        = local.standard_realm_name
+  saml_entity_id             = "https://dev.loginproxy.gov.bc.ca/auth/realms/_bceidbasicbusiness/"
+  single_sign_on_service_url = local.siteminder_single_sign_on_service_url
+  signing_certificate        = var.siteminder_signing_certificate
+  depends_on = [
+    module.bceidbusiness
+  ]
+}
+
+module "standard_clients" {
+  source            = "./standard-clients"
+  standard_realm_id = module.standard.realm_id
+}
+
+module "master_idir_link" {
+  source           = "github.com/bcgov/social-access-portal-terraform-modules?ref=3066d73/modules/master-idp-link"
+  keycloak_url     = var.keycloak_url
+  idp_realm_id     = module.idir.realm_id
+  idp_realm_name   = module.idir.realm_name
+  idp_public_attrs = ["display_name", "idir_user_guid", "idir_username"]
+}
+
+module "master_viewer_role" {
+  source      = "github.com/bcgov/social-access-portal-terraform-modules?ref=3066d73/modules/master-viewer-role"
+  realm_names = ["master", "standard", "idir", "azureidir", "bceidbasic", "bceidbusiness", "bceidboth"]
+
+  depends_on = [
+    module.standard,
+    module.idir,
+    module.azureidir,
+    module.bceidbasic,
+    module.bceidbusiness,
+    module.bceidboth,
+  ]
+}
```

Comment:
This is great!
I wish all Keycloak configs were defined in terraform as code. Much easier to understand how it works.